django/contrib/gis/gdal/raster/source.py

```diff
@@ -66,13 +73,39 @@ def __init__(self, ds_input, write=False):
 
         # If input is a valid file path, try setting file as source.
         if isinstance(ds_input, str):
-            if not os.path.exists(ds_input):
-                raise GDALException('Unable to read raster source input "{}"'.format(ds_input))
+            if not ds_input.startswith(VSI_FILESYSTEM_BASEPATH) and not os.path.exists(ds_input):
+                raise GDALException('Opening rasters from vsi file paths is not allowed.')
```

Comment:
Thanks for spotting this, the error message like this does not make sense. I had a previous version where direct opening of vsi paths was not allowed. But it should work like files, its the user's responsibility when opening the same vsi file twice.

I actually think this check is now not necessary anymore, this "file exists" check was created when only str input was allowed.  The try/except error message should be explicit enough. So I propose to remove those two lines entirely. What do you think?